src/sentry/api/endpoints/organization_spans_trace.py

```diff
@@ -0,0 +1,100 @@
+import sentry_sdk
+from django.http import HttpRequest, HttpResponse
+from rest_framework.response import Response
+
+from sentry import features
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.bases import NoProjects, OrganizationEventsV2EndpointBase
+from sentry.api.utils import handle_query_errors, update_snuba_params_with_timestamp
+from sentry.models.organization import Organization
+from sentry.search.events.builder import SpansIndexedQueryBuilder
+from sentry.snuba.dataset import Dataset
+from sentry.snuba.referrer import Referrer
+from sentry.utils.validators import INVALID_ID_DETAILS, is_event_id
+
+
+class OrganizationSpansTraceEndpoint(OrganizationEventsV2EndpointBase):
+    publish_status = {
+        "GET": ApiPublishStatus.PRIVATE,
+    }
+
+    def query_trace_data(self, params, trace_id):
+        builder = SpansIndexedQueryBuilder(
+            Dataset.SpansIndexed,
+            params,
+            query=f"trace:{trace_id}",
+            selected_columns=[
```

Comment:
this query will be quite slow because of the amount of columns scanned. Just as an FYI. I know you said you'll pair it down later

Reply:
:+1: yeah I think the endstate is something like 1 query that only grabs the minimum columns (timestamp, parent, name) and secondary queries that pull details for a single transaction.

But I'm curious to see if this "give me everything!" approach is feasible, and this endpoint is behind a feature flag so should have minimal impact

Reply:
@wmak have you tried this query in the admin tool to see how it performs yet?

Reply:
Not yet, I can give that a try

Reply:
Huh nice, "only" 2.5s for a trace with 6500 spans (97 txn) but actually going to remove all the extra columns for now so FE can just focus on rendering first

Reply:
I would note too this basically isn't using the primary index at all (I think). So the performance will only improve.